src-tauri/src/main.rs

```diff
@@ -19,6 +23,48 @@ fn get_shared_daemon() -> SharedServiceDaemon {
     Arc::new(Mutex::new(daemon))
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct ResolvedService {
+    instance_name: String,
+    hostname: String,
+    port: u16,
+    addresses: HashSet<IpAddr>,
+}
+
+#[tauri::command]
+fn resolve_service(service_type: String, state: State<Daemon>) -> Vec<ResolvedService> {
```

Comment:
Perhaps use a hash set to avoid duplicates?
```suggestion
fn resolve_service(service_type: String, state: State<Daemon>) -> HashSet<ResolvedService> {
```

Reply:
#30